### ISSUE TITLE
为什么将NER模型部署到TensorFlow Serving调用的时候输出只有97个字符？

### ISSUE THREAD

**Issue by xxxxxxxxy** [BrikerMan/Kashgari#177]
使用`loaded_model.predict(test_x[:10])`可以输出全部的标签和实体，但是用TensorFlow Serving的时候只输出了97个，似乎是process_x_dataset的作用，但是我又在这个函数里找不到长度的限制。

**Comment by BrikerMan** [comment]
麻烦确认以下几件事情：

1. 模型初始化时候的序列长度
2. 模型初始化代码
3. `utils.load_processor` 加载的 processor 的 `add_bos_eos` 属性的值和 `sequence_length` 属性的值

**Comment by xxxxxxxxy** [comment]
感谢指导，我去一一check~

**Comment by xxxxxxxxy** [comment]
我还有一个疑惑，明明模型保存的时候打了signature，可是我却找不到您具体在哪里打了signature，望指点

**Comment by BrikerMan** [comment]
模型保存成 tf 模型时候其实没有打 signature，就用了默认设置。

**Comment by xxxxxxxxy** [comment]
感谢指导，问题已解决，将模型初始化时候的长度改成可变即可~

**Comment by btb16137** [comment]
你好，想问一下模型初始化时候的长度是在哪里修改呢？有没有说明文档？不是很明白如何修改，我现在做ner返回的标签会在结尾比传入的语句多出来很多[PAD]，我猜测可能是这个原因造成的

**Comment by btb16137** [comment]
你好，想问一下模型初始化时候的长度是在哪里修改呢？有没有说明文档？不是很明白如何修改，我现在做ner返回的标签会在结尾比传入的语句多出来很多[PAD]，我猜测可能是这个原因造成的。可以麻烦帮我解答一下吗？感谢 @BrikerMan @xxxxxxxxy

**Comment by BrikerMan** [comment]
> 你好，想问一下模型初始化时候的长度是在哪里修改呢？有没有说明文档？不是很明白如何修改，我现在做ner返回的标签会在结尾比传入的语句多出来很多[PAD]，我猜测可能是这个原因造成的。可以麻烦帮我解答一下吗？感谢 @BrikerMan @xxxxxxxxy

在初始化 Embedding 层时指定长度，比如用 BERT 时候设定 sequence_length = 100 如下。

```python
import kashgari
from kashgari.tasks.labeling import BLSTMModel
from kashgari.embeddings import BERTEmbedding

bert_embed = BERTEmbedding('<PRE_TRAINED_BERT_MODEL_FOLDER>',
                           task=kashgari.LABELING,
                           sequence_length=100)
model = BLSTMModel(bert_embed)
model.fit(train_x, train_y, valid_x, valid_y)
```

有很多 padding 输出，是因为还没训练好，所以标签打错了。当你序列长度小于你的 maxlen 时候，你拿到的标注长度应该和输入序列长度一致。

**Comment by btb16137** [comment]
感激不尽~我去看一下
